lib/ansible/template/__init__.py

```diff
@@ -482,6 +499,16 @@ def template(self, variable, convert_bare=False, preserve_trailing_newlines=True
             else:
                 return variable
 
+    def is_template(self, data):
+        ''' lets us know if data has a template'''
+        try:
+            new = self.template(data)
+        except UndefinedError:
+            return True
+        except:
+            return False
+        return (new != data)
```

Comment:
I added a couple of test cases to https://github.com/bcoca/ansible/compare/unsafe_unsafe_again...alikins:test_is_template that include passing in a tuple as 'data', and that seems to cause is_template to always return True

```
_______________________________________________________________________ TestTemplarIsTemplate.test_is_template_just_a_tuple _______________________________________________________________________
self = <units.template.test_templar.TestTemplarIsTemplate testMethod=test_is_template_just_a_tuple>

    def test_is_template_just_a_tuple(self):
        some_obj = ('foo', 'bar', 1, 3, 5, None)
        res = self.templar.is_template(some_obj)
>       self.assertFalse(res, 'is_template(%s) return should be False' % repr(some_obj))
E       AssertionError: is_template(('foo', 'bar', 1, 3, 5, None)) return should be False

test/units/template/test_templar.py:347: AssertionError
_________________________________________________________________ TestTemplarIsTemplate.test_is_template_just_a_tuple_with_false __________________________________________________________________
self = <units.template.test_templar.TestTemplarIsTemplate testMethod=test_is_template_just_a_tuple_with_false>

    def test_is_template_just_a_tuple_with_false(self):
        some_obj = (False, 'foo', 'bar', 1, 3, 5, None)
        res = self.templar.is_template(some_obj)
>       self.assertFalse(res, 'is_template(%s) return should be False' % repr(some_obj))
E       AssertionError: is_template((False, 'foo', 'bar', 1, 3, 5, None)) return should be False

test/units/template/test_templar.py:352: AssertionError
__________________________________________________________________ TestTemplarIsTemplate.test_is_template_just_a_tuple_with_true __________________________________________________________________
self = <units.template.test_templar.TestTemplarIsTemplate testMethod=test_is_template_just_a_tuple_with_true>

    def test_is_template_just_a_tuple_with_true(self):
        some_obj = (True, 'foo', 'bar', 1, 3, 5, None)
        res = self.templar.is_template(some_obj)
>       self.assertFalse(res, 'is_template(%s) return should be False' % repr(some_obj))
E       AssertionError: is_template((True, 'foo', 'bar', 1, 3, 5, None)) return should be False

test/units/template/test_templar.py:357: AssertionError
________________________________________________________________ TestTemplarIsTemplate.test_is_template_tuple_with_unsafe_template ________________________________________________________________
self = <units.template.test_templar.TestTemplarIsTemplate testMethod=test_is_template_tuple_with_unsafe_template>

    def test_is_template_tuple_with_unsafe_template(self):
        some_obj = ('foo', 'bar', wrap_var('{{ some_var }}'))
        res = self.templar.is_template(some_obj)
>       self.assertFalse(res, 'is_template(%s) return should be False' % repr(some_obj))
E       AssertionError: is_template(('foo', 'bar', u'{{ some_var }}')) return should be False

test/units/template/test_templar.py:367: AssertionError
```